lib/bson.js

```diff
@@ -1,36 +1,48 @@
 'use strict';
 
-var Map = require('./map'),
-  Long = require('./long'),
-  Double = require('./double'),
-  Timestamp = require('./timestamp'),
-  ObjectID = require('./objectid'),
-  BSONRegExp = require('./regexp'),
-  Symbol = require('./symbol'),
-  Int32 = require('./int_32'),
-  Code = require('./code'),
-  Decimal128 = require('./decimal128'),
-  MinKey = require('./min_key'),
-  MaxKey = require('./max_key'),
-  DBRef = require('./db_ref'),
-  Binary = require('./binary');
+const Buffer = require('buffer').Buffer;
+const Map = require('./map');
+const Long = require('./long');
+const Double = require('./double');
+const Timestamp = require('./timestamp');
+const ObjectID = require('./objectid');
+const BSONRegExp = require('./regexp');
+const Symbol = require('./symbol');
+const Int32 = require('./int_32');
+const Code = require('./code');
+const Decimal128 = require('./decimal128');
+const MinKey = require('./min_key');
+const MaxKey = require('./max_key');
+const DBRef = require('./db_ref');
+const Binary = require('./binary');
 
 // Parts of the parser
-var deserialize = require('./parser/deserializer'),
-  serializer = require('./parser/serializer'),
-  calculateObjectSize = require('./parser/calculate_size');
+const internalDeserialize = require('./parser/deserializer');
+const internalSerialize = require('./parser/serializer');
+const internalCalculateObjectSize = require('./parser/calculate_size');
+const ensureBuffer = require('./ensure_buffer');
 
 /**
  * @ignore
  * @api private
  */
 // Default Max Size
-var MAXSIZE = 1024 * 1024 * 17;
+const MAXSIZE = 1024 * 1024 * 17;
 
 // Current Internal Temporary Serialization Buffer
-var buffer = new Buffer(MAXSIZE);
+let buffer = new Buffer(MAXSIZE);
```

Comment:
Are we allocating 17 MB of data here?

Reply:
We are! This is actually in line with the server code, and generally is over 16MB so we can accommodate the over head of message envelopes to the server, particular bulk cases

Reply:
It seems like  a lot to preallocate. Especially on the web, when even the heaviest web pages are usually around 40Mb of heap space.

Reply:
That's a good point, especially considering this is brought in as a dependency for e.g. stitch's js SDK. I've created [NODE-1599](https://jira.mongodb.org/browse/NODE-1599) to track this.

Reply:
okay fixed